Delete a listed list throws a NullPointerException

### Description
After performing a list command, deleting the content of the displayed list throws an error that is not handled.

### Steps for reproducing
Assuming the sample data is used, execute:
```
list /mod CS2040S
delete CS2040S /r
```

### Expected output
Shows an empty list and a message success to the user.

### Actual output
Does nothing in UI and an unhandled exception is thrown viewable from the terminal.

### Other details
None

### Error output
```
Exception in thread "JavaFX Application Thread" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1787)
        at javafx.fxml.FXMLLoader$ControllerMethodEventHandler.handle(FXMLLoader.java:1670)
        at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
        at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
        at javafx.event.Event.fireEvent(Event.java:198)
        at javafx.scene.Node.fireEvent(Node.java:8879)
        at com.sun.javafx.scene.control.behavior.TextFieldBehavior.fire(TextFieldBehavior.java:184)
        at com.sun.javafx.scene.control.behavior.TextInputControlBehavior.lambda$keyMapping$62(TextInputControlBehavior.java:330)
        at com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
        at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
        at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
        at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
        at javafx.event.Event.fireEvent(Event.java:198)
        at javafx.scene.Scene$KeyHandler.process(Scene.java:4058)
        at javafx.scene.Scene$KeyHandler.access$1500(Scene.java:4004)
        at javafx.scene.Scene.processKeyEvent(Scene.java:2121)
        at javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2595)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:217)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:149)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:248)
        at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:390)
        at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:247)
        at com.sun.glass.ui.View.handleKeyEvent(View.java:547)
        at com.sun.glass.ui.View.notifyKey(View.java:971)
        at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
        at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.sun.javafx.reflect.Trampoline.invoke(MethodUtil.java:76)
        at jdk.internal.reflect.GeneratedMethodAccessor8.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.sun.javafx.reflect.MethodUtil.invoke(MethodUtil.java:273)
        at com.sun.javafx.fxml.MethodHelper.invoke(MethodHelper.java:83)
        at javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1784)
        ... 47 more
Caused by: java.lang.NullPointerException
        at seedu.address.model.ModelManager.updateFilteredLectureList(ModelManager.java:281)
        at seedu.address.logic.trackereventsystem.observers.ListObserver.onModuleEdited(ListObserver.java:92)
        at seedu.address.logic.trackereventsystem.TrackerEventSystem.triggerOnModuleEditedEvent(TrackerEventSystem.java:109)
        at seedu.address.logic.LogicManager.triggerTrackerEvents(LogicManager.java:156)
        at seedu.address.logic.LogicManager.execute(LogicManager.java:81)
        at seedu.address.ui.MainWindow.executeCommand(MainWindow.java:208)
        at seedu.address.ui.CommandBox.handleCommandEntered(CommandBox.java:59)
        ... 58 more
```

--- a/src/main/java/seedu/address/logic/trackereventsystem/observers/ListObserver.java
+++ b/src/main/java/seedu/address/logic/trackereventsystem/observers/ListObserver.java
@@ -47,14 +47,13 @@ public class ListObserver implements
                 && editedInLectureName.equals(curLecture.getName());
     }
 
-
     @Override
     public void onLectureEdited(ModuleCode moduleCode, ReadOnlyLecture originalLecture, ReadOnlyLecture editedLecture) {
         ReadOnlyModule curModule = model.getListedLecturesByModule();
         ReadOnlyLecture curLecture = model.getListedVideosByLecture();
         DisplayListLevel curDisplayListLevel = model.getLastListLevel();
 
-        if (isLecturesAffectedByLectureEdit(curDisplayListLevel, curModule, curLecture, originalLecture, moduleCode)) {
+        if (isLecturesAffectedByLectureEdit(curDisplayListLevel, curModule, originalLecture, moduleCode)) {
             model.updateFilteredLectureList(new LecturePredicate(curModule), curModule);
         }
 
@@ -65,7 +64,7 @@ public class ListObserver implements
 
     private boolean isLecturesAffectedByLectureEdit(DisplayListLevel curDisplayListLevel,
             ReadOnlyModule curModule, ReadOnlyLecture curLecture,
-            ReadOnlyLecture originalLecture, ModuleCode editedInModuleCode) {
+            ModuleCode editedInModuleCode) {
         return curDisplayListLevel == DisplayListLevel.LECTURE
                 && editedInModuleCode.equals(curModule.getCode());
     }
@@ -89,10 +88,18 @@ public class ListObserver implements
         }
 
         if (isLecturesAffectedByModuleEdit(curDisplayListLevel, curModule, originalModule)) {
+            if (editedModule == null) { // Module is deleted
+                model.updateAllFilteredListAsHidden();
+                return;
+            }
             model.updateFilteredLectureList(new LecturePredicate(editedModule), editedModule);
         }
 
         if (isVideosAffectedByModuleEdit(curDisplayListLevel, curModule, originalModule)) {
+            if (editedModule == null) { // Module is deleted
+                model.updateAllFilteredListAsHidden();
+                return;
+            }
             model.updateFilteredVideoList(new VideoPredicate(curLecture), editedModule.getCode(), curLecture);
         }
     }

--- a/src/main/java/seedu/address/model/Model.java
+++ b/src/main/java/seedu/address/model/Model.java
@@ -303,6 +303,11 @@ public interface Model {
             ModuleCode moduleCode, ReadOnlyLecture lecture);
 
     /**
+     * Updates the filter to hide all the contents of a list.
+     */
+    void updateAllFilteredListAsHidden();
+
+    /**
      * Returns the last list type to displayed to user.
      */
     DisplayListLevel getLastListLevel();

--- a/src/main/java/seedu/address/model/ModelManager.java
+++ b/src/main/java/seedu/address/model/ModelManager.java
@@ -226,6 +226,17 @@ public class ModelManager implements Model {
     //=========== Filtered List Accessors =============================================================
 
     @Override
+    public void updateAllFilteredListAsHidden() {
+        filteredModules.setPredicate(PREDICATE_HIDE_ALL_MODULES);
+        if (filteredLectures != null) {
+            filteredLectures.setPredicate(PREDICATE_HIDE_ALL_LECTURES);
+        }
+        if (filteredVideos != null) {
+            filteredVideos.setPredicate(PREDICATE_HIDE_ALL_VIDEOS);
+        }
+    }
+
+    @Override
     public DisplayListLevel getLastListLevel() {
         return lastListLevel;
     };

--- a/src/test/java/seedu/address/testutil/ModelStub.java
+++ b/src/test/java/seedu/address/testutil/ModelStub.java
@@ -192,6 +192,11 @@ public class ModelStub implements Model {
     }
 
     @Override
+    public void updateAllFilteredListAsHidden() {
+        throw new AssertionError("This method should not be called.");
+    }
+
+    @Override
     public DisplayListLevel getLastListLevel() {
         throw new AssertionError("This method should not be called.");
     }

